src/Articles.jsx

```diff
@@ -0,0 +1,23 @@
+import { useEffect, useState } from "react";
+import ArticleList from "./ArticleList";
+import { getArticles } from "../utils/api";
+
+const Articles = () => {
+  const [articles, setArticles] = useState([]);
+
+useEffect(() => {
+    getArticles().then(({ articles }) => {
+        setArticles(articles)
+    })
+}, [articles])
```

Comment:
just to confirm you understand what this useEffect does.

It triggers on mount and makes an api request.

It resolves a promise and sets the state (`setArticles`) to the result.

Since the value of `articles` has changed, it triggers the useEffect and makes an api request.

It resolves a promise and sets the state (`setArticles`) to the result.

Since the value of `articles` has changed, it triggers the useEffect and makes an api request.

It resolves a promise and sets the state (`setArticles`) to the result.

Since the value of `articles` has changed, it triggers the useEffect and makes an api request.

...


I hope you understand what I'm getting at.